### PR TITLE
Tests for RHEL-17494

### PR DESCRIPTION
**Pull request by ppisar**
Upstream commit: 8234f4221efee5269258b384cffb3ac31107aa15
Upstream commit: a7a2c93eb7c2417ff7fc0838884ee20ff549db0c
For: https://issues.redhat.com/browse/RHEL-17494
For: https://github.com/rpm-software-management/libdnf/pull/1657

@jan-kolarik, please review this packport of your commits into rhel-9.5.0 branch.

### CODE REVIEW

**Comment by ppisar** [comment]
There is a regression in https://github.com/rpm-software-management/libdnf/pull/1657. Putting this ci-dnf-stack request on hold.

**Review by jan-kolarik — APPROVED** [review]

**Comment by jan-kolarik** [comment]
> There is a regression in [rpm-software-management/libdnf#1657](https://github.com/rpm-software-management/libdnf/pull/1657). Putting this ci-dnf-stack request on hold.

I wanted to add a test covering the detected regression, though I couldn't reproduce it by creating a testing scenario in CI stack. Even at runtime it's not always reproducible.

**Comment by ppisar** [comment]
I understand. I will merge these tests after building dnf as I we already have new dnf tests in this branch.

**Pull request merged**
